### PR TITLE
Webservice - catalogs export

### PR DESCRIPTION
**Pull request by roxanadina**
Open an endpoint with all enums used in the system

### CODE REVIEW

File: webservice/export-catalogs.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+		catalogItem.options = extension.members.toArray().map(function (option) {
+			return { name: option, label: extension.meta[option].label };
+		});
+		return catalogItem;
+	});
+	[].push.apply(catalogs, enums);
+
+	// services
+	var services = db.BusinessProcess.extensions.toArray().map(function (bp) {
+		return { name: bp.__id__, label: bp.label };
```
Comment (mtuchowski) [inline webservice/export-catalogs.js]: Labels are on `bp.prototype.label`.

File: webservice/export-catalogs.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	});
+	catalogs.push({ services: services });
+
+	// registrations
+	var registrations = db.Registration.extensions.toArray().map(function (reg) {
```
Comment (mtuchowski) [inline webservice/export-catalogs.js]: We don't (most of the time) create a classes for registrations (for example https://github.com/egovernment/eregistrations-guatemala/blob/master/model/business-process-company/registrations/business-patent.js#L7-L9) so this array is empty in GT. They are, in that case, available under `db.Registration.instances` but their `__id__` is not human friendly (`BusinessProcessMandate#/registrations/map/mandateInscription`). Using `reg.key` will help but may cause duplicates in Salvador.

Using both methods, `extensions` and `instances` will also cause duplicates as for each class we will have it's instances for each BP that uses it.

This have to be rethought…

File: webservice/export-catalogs.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	catalogs.push({ registrations: registrations });
+
+	// documents
+	var documents = db.Document.extensions.toArray().map(function (doc) {
+		return { name: doc.__id__, label: doc.label };
+	});
+	catalogs.push({ documents: documents });
+
+	// costs
+	var costs = db.Cost.extensions.toArray().map(function (cost) {
```
Comment (mtuchowski) [inline webservice/export-catalogs.js]: Same case as in registrations.

File: webservice/export-catalogs.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+	var enums = db.StringLine.extensions.toArray().filter(function (extension) {
+		return extension.meta && extension.members;
+	}).map(function (extension) {
+		var catalogItem = { name: extension.__id__ };
+		catalogItem.options = extension.members.toArray().map(function (option) {
+			return { name: option, label: extension.meta[option].label };
+		});
+		return catalogItem;
+	});
+	[].push.apply(catalogs, enums);
```
Comment (mtuchowski) [inline webservice/export-catalogs.js]: Let's maybe add those enum types under `catalogs.enums` in line with `catalogs.services` etc.

Comment (roxanadina) [inline webservice/export-catalogs.js]: @mtuchowski  they all are enums for the end consumer
i don't think this new grouping brings any extra value

Comment (kamsi) [inline webservice/export-catalogs.js]: @roxanadina maybe it does not bring much semantically, but still @mtuchowski has a point here. It's like we use namespace for our important groups (`services`, `documents` etc.) while flood the catalogs with all others directly. In the worst case scenario we can hit a name collision, when I for example have an enum entry `services`. So we should group them at least for the purpose of avoiding that (still there will be possibility of name collision between enum entries themselves).

**Comment by roxanadina** [comment]
@kamsi, @mtuchowski 
enums added under "enums"
roles also added

File: webservice/export-catalogs.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+			return { name: option, label: extension.meta[option].label };
+		});
+		return catalogItem;
+	});
+	catalogs.push({ enums: enums });
+
+	// services + roles
+	var services = [];
+	var roles = [];
+	var bpRoles = {};
```
Comment (kamsi) [inline webservice/export-catalogs.js]: I really thinks we should not encourage programmers to identify processingSteps with roles. At some point we might be asked to expose roles, and it is something different than steps. Let's stick to processingSteps.

Comment (roxanadina) [inline webservice/export-catalogs.js]: @kamsi  true! my bad

**Pull request merged**